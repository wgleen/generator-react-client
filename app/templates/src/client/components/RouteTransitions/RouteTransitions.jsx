import React from 'react'
import PropTypes from 'prop-types'
import { AnimatedSwitch } from 'react-router-transition'
import './routeTransitions.scss'

const RouteTransitions = (props) => {
  const { children } = props

  return (
    <AnimatedSwitch
      atEnter={{ opacity: 0 }}
      atLeave={{ opacity: 0 }}
      atActive={{ opacity: 1 }}
      className='route-wrapper'
    >
      {children}
    </AnimatedSwitch>
  )
}

RouteTransitions.defaultProps = {
  children: null
}

RouteTransitions.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.string
  ])
}

export default RouteTransitions
