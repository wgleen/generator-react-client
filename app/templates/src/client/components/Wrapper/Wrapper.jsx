import React from 'react'
import { Provider } from 'react-redux'
import {
  store
} from '../../lib/store'

const Wrapper = props => (
  <Provider store={store}>
    {props.children}
  </Provider>
)

export default Wrapper