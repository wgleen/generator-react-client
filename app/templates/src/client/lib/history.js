import { createBrowserHistory } from 'history'

const _history = createBrowserHistory()

_history.listen(() => {
  window.scrollTo(0, 0)
})

export default _history