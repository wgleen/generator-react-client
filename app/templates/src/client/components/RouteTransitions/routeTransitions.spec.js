import React from 'react'
import { mount } from 'enzyme'
import history from '../../lib/history'
import Wrapper from '../Wrapper/Wrapper'
import { ConnectedRouter } from 'connected-react-router'
import RouteTransitions from '../RouteTransitions/RouteTransitions'

describe('<RouteTransitions />', () => {
  it('should render correctly', () => {
    let component = mount(
      <Wrapper>
        <ConnectedRouter history={history}>
          <RouteTransitions />
        </ConnectedRouter>
      </Wrapper>
    )

    expect(component).toMatchSnapshot()
    
    component.unmount()
  })
})