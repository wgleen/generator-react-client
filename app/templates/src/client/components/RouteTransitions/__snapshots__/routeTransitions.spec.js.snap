// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteTransitions /> should render correctly 1`] = `
<Wrapper>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(ConnectedRouter)
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ConnectedRouter
        action="POP"
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        onLocationChanged={[Function]}
      >
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <RouteTransitions>
            <RouteWrapper
              atActive={
                Object {
                  "opacity": 1,
                }
              }
              atEnter={
                Object {
                  "opacity": 0,
                }
              }
              atLeave={
                Object {
                  "opacity": 0,
                }
              }
              className="route-wrapper"
            >
              <Route>
                <AnimatedSwitch
                  atActive={
                    Object {
                      "opacity": 1,
                    }
                  }
                  atEnter={
                    Object {
                      "opacity": 0,
                    }
                  }
                  atLeave={
                    Object {
                      "opacity": 0,
                    }
                  }
                  className="route-wrapper"
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                >
                  <RouteTransition
                    atActive={
                      Object {
                        "opacity": 1,
                      }
                    }
                    atEnter={
                      Object {
                        "opacity": 0,
                      }
                    }
                    atLeave={
                      Object {
                        "opacity": 0,
                      }
                    }
                    className="route-wrapper"
                    mapStyles={[Function]}
                    runOnMount={false}
                    wrapperComponent="div"
                  >
                    <TransitionMotion
                      defaultStyles={null}
                      didLeave={[Function]}
                      styles={
                        Array [
                          Object {
                            "data": <Switch
                              location={
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                }
                              }
                            />,
                            "key": "",
                            "style": Object {
                              "opacity": Object {
                                "damping": 26,
                                "precision": 0.01,
                                "stiffness": 170,
                                "val": 1,
                              },
                            },
                          },
                        ]
                      }
                      willEnter={[Function]}
                      willLeave={[Function]}
                    >
                      <div
                        className="route-wrapper"
                      >
                        <div
                          key=""
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <Switch
                            key=""
                            location={
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              }
                            }
                          />
                        </div>
                      </div>
                    </TransitionMotion>
                  </RouteTransition>
                </AnimatedSwitch>
              </Route>
            </RouteWrapper>
          </RouteTransitions>
        </Router>
      </ConnectedRouter>
    </Connect(ConnectedRouter)>
  </Provider>
</Wrapper>
`;
